### ISSUE TITLE
Adversarial Examples Are Not Bugs, They Are Features

### ISSUE THREAD

**Issue by josejhlee** [josejhlee/doodling#1]
![image](https://user-images.githubusercontent.com/70194535/130149235-c8d2dddc-c835-4865-832a-4ef0844ff49a.png)
link: https://arxiv.org/abs/1905.02175

### Adversarial attacks
![image](https://user-images.githubusercontent.com/70194535/130174221-98e02508-09fe-498e-8e10-a399f1bd0940.png)

### Standard training vs Robust training
Standard training
![image](https://user-images.githubusercontent.com/70194535/130158160-8afa9b1e-f843-4113-b1a1-9623803e056e.png)
Robust training (adversarial loss function)
![image](https://user-images.githubusercontent.com/70194535/130158169-09f403c3-d8be-45be-b52e-90d4c76b7e1d.png) 
### What was the common consensus on adversarial attacks?
- Viewed adversarial examples as aberrations arising either from the high dimensional nature of the input space or statistical fluctuations in the training data.
### What this paper proposes?
- Adversarial vulnerability is a direct result of our models' sensitivity to well-generalization features in the data. 
> The model's main objective is to increase the accuracy rate. If there is a feature that cooperates with that goal, it will make use of it. Because non-robust features are well-generalizing features, enhancing the accuracy, it is exploited. This explains why different models have adversarial transferability. The two models were trying to find the best classifying features.

![image](https://user-images.githubusercontent.com/70194535/130163596-6f9fc0fb-afa9-4df2-a1ee-c388048c447e.png)

## defining robustness
Binary classification (+1 or -1)
![image](https://user-images.githubusercontent.com/70194535/130155002-d36f32e4-c08b-4e01-813d-f7b8b6f55a9b.png)

### p-useful features
![image](https://user-images.githubusercontent.com/70194535/130153027-a58d0c48-54db-466d-9ce6-7fc540b09d3e.png)
> As the name implies, we define the features that are useful for classification.

### Gamma-robustly useful features
![image](https://user-images.githubusercontent.com/70194535/130153048-b2f36b9a-0d26-4a0c-8a62-921393d995b0.png)
> Even with the introduction of perturbation, it remains gamma-useful feature.

### non-robust useful feature
- A useful, non-robust feature is a feature which is p-useful for some p
bounded away from zero, but is not a gamma-robust feature for any gamma >= 0.
> That means that while under no perturbation a feature was giving a positive value for a given x, it may give a negative value for the same x under perturbation application.  

### Constructing a robust dataset (with only robust features)
- Making robust images directly from the original pictures by minimizing the distance between the features.
- Start from random noise (or another random image from another class) and backpropagate wrt the image. 
![image](https://user-images.githubusercontent.com/70194535/130164437-0d71b52a-aab0-4cec-9dc9-fc0cbe3fc1f9.png)
### Constructing a non-robust dataset (with only non-robust features)

![image](https://user-images.githubusercontent.com/70194535/130166033-07dec13b-e373-4dda-ad8c-c2deeb284c94.png)

## Results of the experiment
![image](https://user-images.githubusercontent.com/70194535/130165692-4b1db2e4-590e-4cd5-81f0-af9f833529f5.png)
